app/models/user.rb

```diff
@@ -19,6 +19,20 @@ class User < ApplicationRecord
 
   after_create :create_cart_list
 
+  def follow(other_user)
+    return if other_user == self
```

Comment:
例外エラーを出してあげた方が良いと思います！
自分をフォローしようとしたら、フロントにエラーを出そうと思いますので、returnをして何もなく終わらせるよりかは、例外を出した方がそのきっかけになります。

```suggestion
    raise ArgumentError if other_user == self
```